Prevent numbers from being formatted in scientific notation in forms

maurizi opened the pull request:
Connects to #2733

jwalgran commented:
The tag is working well for the built in fields but I added a custom `Decimal` field and the view mode value is getting scientifically notated.

Looks good in the form

<img width="548" alt="screen shot 2017-08-31 at 2 42 47 pm" src="https://user-images.githubusercontent.com/17363/29946703-2bd70868-8e5b-11e7-87bb-37ee038dae48.png">

Notated after save

<img width="368" alt="screen shot 2017-08-31 at 2 42 06 pm" src="https://user-images.githubusercontent.com/17363/29946719-3eb0e544-8e5b-11e7-93c4-c19b1949465a.png">

maurizi commented:
> Custom decimal field values in the view mode are getting scientifically notated.
Fixed in (rebased) commit https://github.com/OpenTreeMap/otm-core/pull/3208/commits/537c391b84f51f672a1c8d49eb7729fa4fd897c8. I also fixed formatting for numbers in the edits sidebar, and made sure we don't show trailing zeros

jwalgran commented on opentreemap/treemap/util.py:
```diff
@@ -231,3 +231,10 @@ def make_udf_name_from_key(key):
+def num_format(num):
+    if isinstance(num, float):
+        # Allow for up to 10 digits of precision, but strip trailing 0s
+        return '{0:.10f}'.format(num).rstrip('0')
```
For any "whole number" value like `123.0` this `rstrip` is leaving a trailing `.`

<img width="533" alt="screen shot 2017-09-06 at 9 55 21 am" src="https://user-images.githubusercontent.com/17363/30124353-bd3e8c80-92e9-11e7-8b35-71f161c852c1.png">
 



<img width="737" alt="screen shot 2017-09-06 at 10 00 55 am" src="https://user-images.githubusercontent.com/17363/30124549-5be49050-92ea-11e7-93f1-3a25655e5643.png">

maurizi replied:
Fixed in https://github.com/OpenTreeMap/otm-core/pull/3208/commits/7bdee5f995cedd6ee22b6f523e475697726045da

maurizi commented on opentreemap/treemap/util.py:
```diff
@@ -231,3 +231,10 @@ def make_udf_name_from_key(key):
+def num_format(num):
+    if isinstance(num, float):
+        # Allow for up to 10 digits of precision, but strip trailing '0' or '.'
+        return '{0:.10f}'.format(num).rstrip('0.')
```
Sigh... sorry for all the back and forth. Fixed in https://github.com/OpenTreeMap/otm-core/pull/3208/commits/149dfeafd83db8fa047dcbeaf5afef674fd40522
